File: frontend/src/types/permission.ts
export interface PermissionTreeNode {
  id: number
  permissionCode: string
  permissionName: string
  permissionType: string
  resourcePath: string
  action: string
  parentId: number
  children: PermissionTreeNode[]
}

export interface RolePermissionBundle {
  id: string
  name: string
  sort: number
  permissionIds: number[]
  permissionCodes: string[]
}

File: src/main/java/com/company/idm/domain/rbac/Permission.java
package com.company.idm.domain.rbac;

import com.company.idm.common.enums.PermissionType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 权限领域实体，描述菜单、按钮或接口权限。
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Permission {

    private Long id;
    private String permissionCode;
    private String permissionName;
    private PermissionType permissionType;
    private String resourcePath;
    private String action;
    private Long parentId;
    private Integer sortNo;
    private Integer status;
    private String remark;
}

